Updated the playbook section in this docs

DesmondSanctity opened the pull request:
##### SUMMARY
This commit contains changes made to the developing_collection_structure.rst file in the playbook section. It is a guidline for playbooks within a collection as stated by @samccann in Issue #77285

Fixes #77285
##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
This is the path to the file that was edited 
https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/dev_guide/developing_collections_structure.rst

##### ADDITIONAL INFORMATION
This PR is set to add more guidline for playbooks within a collection and how to run them without possible errors

```

DesmondSanctity committed:
Updated the playbook section in this docs

This commit contains changes made to the developing_collection_structure.rst file in the playbook section. It is a guidline for playbooks within a collection as stated by @samccann in Issue #77285

samccann commented on docs/docsite/rst/dev_guide/developing_collections_structure.rst:
```diff
@@ -185,6 +185,13 @@ This will keep the playbook in 'collection context', as if you had added ``colle
+Also, playbooks within a collection follow the same guidelines as any playbooks except for these few adjustments:
+
+ - Directory: It must be in ``/playbooks directory``.
+ - Hosts: The host should be defined as a variable so the users of a playbook do not mistakenly run the plays against their entire inventory (if the host is set to all) e.g. ``hosts: '{{target|default("all")}}'``.
```
```suggestion
 - Hosts: The host should be defined as a variable so the users of a playbook do not mistakenly run the plays against their entire inventory (if the host is set to all). For example -  ``hosts: '{{target|default("all")}}'``.
```

samccann commented on docs/docsite/rst/dev_guide/developing_collections_structure.rst:
```diff
@@ -185,6 +185,13 @@ This will keep the playbook in 'collection context', as if you had added ``colle
+ - Directory: It must be in ``/playbooks directory``.
+ - Hosts: The host should be defined as a variable so the users of a playbook do not mistakenly run the plays against their entire inventory (if the host is set to all) e.g. ``hosts: '{{target|default("all")}}'``.
+
+To run the plays, users can now use such command as ``ansible-playbook --e 'targets=webservers'`` or ``ansible-playbook --limit webservers``. Either way, the collection owner should document their playbooks and how to use them in the ``/docs`` folder or ``README`` file.
```
```suggestion
To run the plays, users can now use such commands as ``ansible-playbook --e 'targets=webservers'`` or ``ansible-playbook --limit webservers``. Either way, the collection owner should document their playbooks and how to use them in the ``/docs`` folder or ``README`` file.
```

DesmondSanctity committed:
grammatical edits to previous doc fix

This is a grammatical error fix to the documentation as suggested in the PR #77747
